### PR TITLE
LPS-174623 LPS-178589 Add reindex alert to the Semantic Search Configuration UI

### PR DESCRIPTION
**Pull request by oliv-yu**
https://issues.liferay.com/browse/LPS-174623 - Add warning to the Semantic Search Configuration UI to indicate that changes to the provider settings requires a reindex

Copied from Acceptance Criteria:

- An info alert is displayed above the "Save" button with the text `Changes to this configuration require a reindex in Control Panel > Search > Index Actions. If types were added or removed, you can reindex just the affected types. Otherwise, a full reindex is required.` for the following conditions:
-- If "Text Embeddings Enabled" is enabled AND an Administrator makes changes to the provider settings (ie, switching provider, model, dimension etc.) or "Types"
-- If "Text Embeddings Enabled" was previously saved as unchecked and then checked
- When "Text Embeddings Enabled" is not checked, the info alert will not be displayed

Dev notes: In order to determine when the alert would show, there's logic to compare the initially computed values with the current values. I couldn't use the `formik.dirty` one since this is specific to certain inputs. Let me know if the changes makes sense, or if it can be improved in any way, thanks!

Screenshot 2023-03-17 at 6 10 28 PM

### CODE REVIEW

**Comment by oliv-yu** [comment]
ci:test:sf

**Comment by oliv-yu** [comment]
ci:test:search

**Review by thektan — CHANGES_REQUESTED** [review]
Hi @oliv-yu, thanks for these changes! Can you make the `useMemo` change and you can run ci:forward afterwards!

File: modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/semantic_search/js/configuration/index.js [hunk 1]
```diff
@@ -41,6 +42,31 @@ const DEFAULT_TEXT_EMBEDDING_PROVIDER_CONFIGURATIONS = {
...
+	if (Array.isArray(item1) && Array.isArray(item2)) {
+		return (
+			item1.length !== item2.length ||
+			item1.some((str) => !item2.includes(str)) ||
+			item2.some((str) => !item1.includes(str))
+		);
+	}
+
+	return item1 !== item2;
+}
```
[the remark this answers is not in the document]
Comment (oliv-yu) [reply modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/semantic_search/js/configuration/index.js:68]: Noted! Will try to be positive next time 😄

**Comment by oliv-yu** [comment]
ci:test:sf

**Comment by oliv-yu** [comment]
ci:test:search

**Comment by oliv-yu** [comment]
ci:forward

**Pull request closed without merging**